ipa/ipa_provider.py

```diff
@@ -372,6 +373,13 @@ def _update_history(self):
             test_log=self.log_file
         )
 
+    def _wait_on_instance(self, state):
+        """Wait until instance is in given state."""
+        current_state = 'Undefined'
+        while state != current_state:
```

Comment:
Loop needs a timed/counted exit condition. This assumes that the instance will reach the expected state. However, there is no gurantee that such state will ever be reached. Instance launch could fail or hang or whatever, for example. Azure is the slowest platform, I'd say if we do not reach the desired state in 5 minutes, throw an exception.